src/resolvers/dns.rs

```diff
@@ -442,7 +442,7 @@ impl Resolver for DnsResolver {
         };
 
         let _send_result = terminate_tx.send(());
-        let _join_result = handle.await;
+        handle.await.expect("Resolver task panicked unexpectedly");
```

Comment:
a bit nitpicky, but i think what we should _really_ be doing here, if you want to propagate the panic in the nicest possible way, is to look inside the `JoinError`, take the panic out, and continue unwinding with the same payload, the way @sunshowers did here:
https://github.com/oxidecomputer/async-bb8-diesel/blob/01be5c0c98eddf0041d04ade5a5ba685a570336a/src/async_traits.rs#L407-L409

Reply:
Done here: https://github.com/oxidecomputer/qorb/pull/48